roda-ui/roda-wui/src/main/java/org/roda/wui/client/portal/BrowseDIPPortal.java

```diff
@@ -0,0 +1,381 @@
+      HistoryUtils.newHistory(BrowseTop.RESOLVER);
+      callback.onSuccess(null);
+    }
+
+    private void load(final Viewers viewers, final List<String> historyTokens, final AsyncCallback<Widget> callback) {
+      if (!historyTokens.isEmpty()) {
+        final String historyDipUUID = historyTokens.get(0);
+        final String historyDipFileUUID = historyTokens.size() > 1 ? historyTokens.get(1) : null;
+        BrowserService.Util.getInstance().retrieveDipBundle(historyDipUUID, historyDipFileUUID,
+          LocaleInfo.getCurrentLocale().getLocaleName(), new AsyncCallback<BrowseDipBundle>() {
+            @Override
+            public void onFailure(Throwable throwable) {
+              AsyncCallbackUtils.defaultFailureTreatment(throwable);
+            }
+
+            @Override
+            public void onSuccess(BrowseDipBundle browseDipBundle) {
+              BrowserService.Util.getInstance().showDIPEmbedded(new AsyncCallback<Boolean>() {
+                @Override
+                public void onFailure(Throwable throwable) {
+                  AsyncCallbackUtils.defaultFailureTreatment(throwable);
+                }
+
+                @Override
+                public void onSuccess(Boolean showEmbedded) {
+                  IndexedDIP dip = browseDipBundle.getDip();
+                  if (StringUtils.isNotBlank(dip.getOpenExternalURL()) && !showEmbedded) {
+                    String url = IndexedDIPUtils.interpolateOpenExternalURL(dip,
+                      LocaleInfo.getCurrentLocale().getLocaleName());
+                    Window.open(url, "_blank", "");
+                    Toast.showInfo(messages.browseFileDipOpenedExternalURL(), url);
+                    History.back();
+                  } else {
+                    callback.onSuccess(new BrowseDIPPortal(viewers, browseDipBundle));
+                  }
+
+                }
+              });
+            }
+          });
+      } else {
+        errorRedirect(callback);
+      }
+    }
+
+  };
+
+  interface MyUiBinder extends UiBinder<Widget, BrowseDIPPortal> {
+  }
+
+  private static MyUiBinder uiBinder = GWT.create(MyUiBinder.class);
+
+  private static ClientMessages messages = GWT.create(ClientMessages.class);
+
+  // IDENTIFICATION
+  @UiField
+  TitlePanel title;
+
+  // DESCRIPTIVE METADATA
+  @UiField
+  HTML descriptiveMetadata;
+
+  @UiField
+  FlowPanel preMetadata;
+
+  // DIP
+  @UiField
+  AccessibleFocusPanel keyboardFocus;
+
+  @UiField
+  FlowPanel center;
+
+  @UiField
+  FlowPanel container;
+
+  public BrowseDIPPortal(Viewers viewers, BrowseDipBundle bundle) {
+    IndexedDIP dip = bundle.getDip();
+    String aipId = dip.getFileIds().get(0).getAipId();
+    String dipTitle = dip.getTitle();
+    DIPFile dipFile = bundle.getDipFile();
+
+    if (aipId == null) {
+      GWT.log("AIP ID não encontrado para DIP " + dip.getId());
+      Toast.showError("Não foi possível obter o AIP associado ao DIP");
```

Comment:
We should always use English in the code, and messages should be managed by ClientMessages and the translations in their respective files.